src/formats/TRR.cpp

```diff
@@ -50,8 +49,9 @@ void TRRFormat::read(Frame& frame) {
     bool has_positions = bool(has_prop & TRR_HAS_POSITIONS);
     bool has_velocities = bool(has_prop & TRR_HAS_VELOCITIES);
 
-    frame.set_step(static_cast<size_t>(md_step));  // actual step of MD Simulation
-    frame.set("time", static_cast<double>(time));  // time in pico seconds
+    frame.set_step(static_cast<size_t>(md_step));     // actual step of MD Simulation
+    frame.set("time", static_cast<double>(time));     // time in pico seconds
+    frame.set("lambda", static_cast<double>(lambda)); // coupling parameter for free energy methods
```

Comment:
> I would pick a more informative name for this.

I haven't worked with free energy calculations yet, so I don't know the nomenclature in this field. `lambda` seems to be the universally used short notation. Alternatively it's called _coupling parameter_ (see [here](https://pdfs.semanticscholar.org/149e/43351c0ccee7b8ad8373aaab2380f9742af0.pdf), [here](http://manual.gromacs.org/documentation/current/reference-manual/algorithms/free-energy-calculations.html) and [here](http://www.bevanlab.biochem.vt.edu/Pages/Personal/justin/gmx-tutorials/free_energy/03_workflow.html)). But maybe it would better to explicitly add free energy to the name: _free energy coupling parameter_?

> Is this lambda parameter always used for free energy simulations? Or can it store other data?

Well, since the format doesn't seem well documented this is hard to say. GROMACS [comments](https://github.com/gromacs/gromacs/blob/a265adaddb3049b854cf659ef089bfc9b2d973b5/src/gromacs/trajectory/trajectoryframe.h#L67) say, that it is for free energy. They don't mention anything else. But I know at least two occasions ([1](https://github.com/gromacs/gromacs/blob/40f1c8eaa2bfa37f7c28d55262d3af7daac1b967/src/gromacs/gmxana/gmx_covar.cpp#L567), [2](https://github.com/gromacs/gromacs/blob/49e196fbd08eb69a72fc5003573e4054542782ac/src/gromacs/gmxana/eigio.cpp#L166)) where it's something else :(
Since the latter ones are marked as _misuse_ I think it is somewhat reasonable for us to expect free energy lambda and name it so.

Reply:
Ok, thanks for the investigation. Another possibility is `trr_lambda`, and then documenting that it is usually used for free energy coupling parameter in the doc.

Reply:
Oh, forgot the doc of this property. Done.